File: 백준/Silver/1929. 소수 구하기/소수 구하기.c
#include <stdio.h>
#include <math.h>
int main(void){
    int fst,second;
    scanf("%d %d",&fst,&second);
    int sie[second+1];
    for(int i=0;i<=second;i++)
        sie[i]=1;
    int sq=sqrt(second);
    for(int i=2;i<=sq+1;i++)
        if(sie[i])
            for(int j=i+i;j<=second;j=j+i)
                    sie[j]=0;

    for(int k=fst;k<=second;k++)
        if(sie[k] && k != 1)
            printf("%d\n",k);

    return 0;
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
fst,second=map(int,input().split())
sieve = [True] * (second+1)
sq=int(second**0.5)
for i in range(2, sq + 1):
    if sieve[i] == True:           # i가 소수인 경우
        for j in range(i+i, second+1, i): # i이후 i의 배수들을 False 판정
            sieve[j] = False
for i in range(fst, second+1):
    if sieve[i] == True and i != 1:
        print(i)